microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/integration/ft/FTMetricTest.java

```diff
@@ -0,0 +1,162 @@
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.commons.io.FileUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.ConfigurationException;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+import io.restassured.http.ContentType;
+
+@Ignore("https://issues.redhat.com/browse/WFLY-12590")
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(SetupTask.class)
+public class FTMetricTest {
+    public static final String INCREMENT_CONFIG_PROPERTY = "dummy.increment";
+    public static final String FAULT_CTL_CONFIG_PROPERTY = "dummy.corrupted";
+    public static final String FAILSAFE_INCREMENT_CONFIG_PROPERTY = "dummy.failsafe.increment";
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    @ArquillianResource
+    URL deploymentUrl;
+
+    private byte[] bytes;
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        return ShrinkWrap.create(WebArchive.class, FTMetricTest.class.getSimpleName() + ".war")
+                .addPackage(FTCustomMetricApplication.class.getPackage())
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @Before
+    public void reload() throws ConfigurationException, InterruptedException, TimeoutException, IOException {
+        new Administration(ManagementClientProvider.onlineStandalone()).reload();
+    }
+
+    @Before
+    public void backup() throws IOException {
+        bytes = FileUtils.readFileToByteArray(SetupTask.propertyFile);
+    }
+
+    @After
+    public void restore() throws IOException {
+        FileUtils.writeByteArrayToFile(SetupTask.propertyFile, bytes);
+    }
+
+    /**
+     * @tpTestDetails Multi-component customer scenario to verify correct behaviour when custom counter metric is used.
+     *                The metric increment depends on CDI bean and MP Config property values. Increment for the metric
+     *                is provided by a fail-safe service. If one provider fails (controlled by MP Config property)
+     *                the fail-safe service fallbacks to another provider.
```

Comment:
To be honest I have a very little idea how this test works. What is "correct behavior"? What is the "Increment"?

Reply:
Increment is a value to be added to the counter. Correct behavior - metric is registered, exposed and incremented as expected. Expected values do not need to be in a rest plan. The code is self-descriptive enough.

Reply:
Fixed.